feat(icon): introduce a new ef-config for ef-icon

Nantawat-Poothong opened the pull request:
## Description

Added an element ef-config to provide the data that consume by other elements

Fixes # (issue)
https://jira.refinitiv.com/browse/STG-423

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

Nantawat-Poothong committed:
feat(icon): introduce a new ef-config for ef-icon

Nantawat-Poothong committed:
chore(icon): typo

Nantawat-Poothong committed:
Merge branch 'v6' into feat/ef-config

Nantawat-Poothong committed:
refactor: debug config context variable

Nantawat-Poothong committed:
Merge remote-tracking branch 'origin/feat/ef-config' into feat/ef-config

Nantawat-Poothong committed:
feat(config): update element api

Tolomeo commented on packages/elements/src/config/elements/config.ts:
```diff
@@ -0,0 +1,47 @@
+
+  @provide({ context: efConfig })
+  @state()
+    config: ConfigType = {};
```
We don't provide a fully structure default config here. Does that create any problems with consumers?

Tolomeo commented on packages/elements/src/icon/index.ts:
```diff
@@ -112,13 +124,22 @@ export class Icon extends BasicElement {
+   * @returns content exists
+   */
+  private get iconMap (): string | null {
+    return this.icon && this.config?.icon?.map[this.icon] || null;
```
This looks too interesting to be good.
Is there a way we can avoid making use of all those optional chaining?

Tolomeo reviewed (commented):
The PR looks nice and the changes are contained, as expected.
We could add tests and expand on the demo page though.
I left a couple (probably related) comments

Nantawat-Poothong committed:
test(config): add test & refactor

Nantawat-Poothong committed:
fix(config): fixes error

refinitiv-ui-dev committed:
Revert "chore(release): publish [skip ci]"

This reverts commit 9c5e19ece59cf12e07ab7593f712b4184298b4af.

refinitiv-ui-dev committed:
chore(release): publish [skip ci]

 - @refinitiv-ui/core@6.5.0-next.0
 - @refinitiv-ui/demo-block@6.2.0-next.0
 - @refinitiv-ui/elemental-theme@6.4.0-next.0
 - @refinitiv-ui/elements@6.9.0-next.0
 - @refinitiv-ui/halo-theme@6.6.0-next.0
 - @refinitiv-ui/solar-theme@6.4.0-next.0
 - @refinitiv-ui/translate@6.1.0-next.0
 - @refinitiv-ui/utils@6.3.0-next.0

Tolomeo commented:
As discussed earlier, there are a couple of concerns still to be fixed:
- **Context component naming**: we are currently making use of ef-config but there are doubts on that. An appreciated alternative seems to be `ef-provider`. Please try to discuss with the extended team on what could be a valid decision for this, but keep in mind that the final naming should reflect the intent of the component to provide shared configurations freely consumed by ef components (for example in the future we may decide to funnel the ui theme through the context).
- **EfConfig properties**: let’s follow already defined best practices to define them. That means we will require the developer to specify `config`.
- **Testing**: the test so far checks for the icon component receiving a context value. That doesn’t reflect the functionality provided though and the test should feed the context with a specified value and then assert on the rendered icon html matching an expected content. The idea is to test the feature using a black box approach.

Nantawat-Poothong committed:
Merge branch 'v6' into feat/ef-config
